### PR TITLE
[Structural] clean and improve restart

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
This is a cleanup PR that is long overdue and can now finally be done since the `ProcessInfo` is being passed to `Initialize`

This should not break change anything but I would like to hear others opinions just in case

Anyway this only affects restarts

see #3348

### CODE REVIEW

File: applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py [hunk 1]
```diff
@@ -217,15 +217,7 @@ def Initialize(self):
...
         mechanical_solution_strategy.SetEchoLevel(self.settings["echo_level"].GetInt())
-        if not self.is_restarted():
-            mechanical_solution_strategy.Initialize()
-        else:
-            # SetInitializePerformedFlag is not a member of SolvingStrategy but
-            # is used by ResidualBasedNewtonRaphsonStrategy.
-            try:
-                mechanical_solution_strategy.SetInitializePerformedFlag(True)
-            except AttributeError:
-                pass
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py:228]: this is the important change
now the Elements/Conditions can check themselves if it is a restart and act accordingly

**Review by loumalouomega — APPROVED** [review]
Just a question, instead of a variable, IS_RESTARTED should be a flag?

**Comment by philbucher** [comment]
> Just a question, instead of a variable, IS_RESTARTED should be a flag?

Probably yes, but I am afraid this would silently break stuff if we are not careful

**Comment by philbucher** [comment]
@rubenzorrilla @sunethwarna I recommend using this also in the FluidApp, as this is a better solution compared to what is currently done

**Pull request merged**